pkg/controller/migration/convert/core.go

```diff
@@ -198,6 +238,222 @@ func handleCore(c *components, install *operatorv1.Installation) error {
 	return nil
 }
 
+func getMinReadySeconds(override comp.ReplicatedPodResourceOverrides) *int32 {
+	if reflect.ValueOf(override).IsNil() {
+		return nil
+	}
+	return override.GetMinReadySeconds()
+}
+
+// migrateMinReadySeconds takes the components and migrates their minReadySeconds values to the installation.
+func migrateMinReadySeconds(c *components, install *operatorv1.Installation) error {
+	// Handle calico-node
+	minReadySeconds := getMinReadySeconds(install.Spec.CalicoNodeDaemonSet)
+	if c.node.Spec.MinReadySeconds > 0 {
+		if minReadySeconds == nil {
+			// minReadySeconds set on the component but not the installation so migrate it over.
+			helpers.EnsureCalicoNodeSpecNotNil(install)
+			install.Spec.CalicoNodeDaemonSet.Spec.MinReadySeconds = &c.node.Spec.MinReadySeconds
+		} else {
+			if *minReadySeconds != c.node.Spec.MinReadySeconds {
+				return ErrIncompatibleMinReadySeconds(ComponentCalicoNode)
+			}
+		}
+	} else {
+		// minReadySeconds is not set on the component, check if it's set on the installation.
+		if minReadySeconds != nil {
+			return ErrIncompatibleMinReadySeconds(ComponentCalicoNode)
+		}
+	}
+
+	// Handle typha
+	if c.typha != nil {
+		minReadySeconds = getMinReadySeconds(install.Spec.TyphaDeployment)
+		if c.typha.Spec.MinReadySeconds > 0 {
+			if minReadySeconds == nil {
+				// minReadySeconds set on the component but not the installation so migrate it over.
+				helpers.EnsureTyphaPodSpecNotNil(install)
+				install.Spec.TyphaDeployment.Spec.MinReadySeconds = &c.typha.Spec.MinReadySeconds
+			} else {
+				if *minReadySeconds != c.typha.Spec.MinReadySeconds {
+					return ErrIncompatibleMinReadySeconds(ComponentTypha)
+				}
+			}
+		} else {
+			// minReadySeconds is not set on the component, check if it's set on the installation.
+			if minReadySeconds != nil {
+				return ErrIncompatibleMinReadySeconds(ComponentTypha)
+			}
+		}
+	}
+
+	// Handle kubecontrollers
+	if c.kubeControllers != nil {
+		minReadySeconds = getMinReadySeconds(install.Spec.CalicoKubeControllersDeployment)
+		if c.kubeControllers.Spec.MinReadySeconds > 0 {
+			if minReadySeconds == nil {
+				// minReadySeconds set on the component but not the installation so migrate it over.
+				helpers.EnsureKubeControllersPodSpecNotNil(install)
+				install.Spec.CalicoKubeControllersDeployment.Spec.MinReadySeconds = &c.kubeControllers.Spec.MinReadySeconds
+			} else {
+				if *minReadySeconds != c.kubeControllers.Spec.MinReadySeconds {
+					return ErrIncompatibleMinReadySeconds(ComponentKubeControllers)
+				}
+			}
+		} else {
+			// minReadySeconds is not set on the component, check if it's set on the installation.
+			if minReadySeconds != nil {
+				return ErrIncompatibleMinReadySeconds(ComponentKubeControllers)
+			}
+		}
+	}
+
+	return nil
+}
+
+func getTolerations(override comp.ReplicatedPodResourceOverrides) []corev1.Toleration {
```

Comment:
This function does not seem very useful to me. In each of the places it is used couldn't the function `GetTolerations()` be called on what is passed into this function?

Reply:
Removed